src/backend/drm/compositor/mod.rs

```diff
@@ -2971,89 +2980,117 @@ where
+                let stride =
+                    width * (format::get_bpp(format).expect("Format with unknown bits per pixel") / 8);
+                let size = stride * height;
+
+                let Ok(pixman_format) = pixman::FormatCode::try_from(format) else {
+                    debug!("No pixman format for {format}");
+                    return None;
+                };
+                match pixman::Image::from_slice_mut(
+                    pixman_format,
+                    width,
+                    height,
+                    unsafe { slice::from_raw_parts_mut(data as *mut u32, size / 4) },
+                    stride,
+                    false,
+                ) {
+                    Ok(image) => Some(PixmanTexture::from(image)),
+                    Err(e) => {
+                        debug!("pixman cursor: {e}");
+                        None
+                    }
+                }
+            }
+        }?;
 
-        let render_res = render();
+        let src = Rectangle::from_loc_and_size(Point::default(), cursor_buffer_size).to_f64();
+        let dst = Rectangle::from_loc_and_size(cursor_plane_location, self.cursor_size);
 
-        // restore the renderer debug flags
-        renderer.set_debug_flags(renderer_debug_flags);
+        let ret = cursor_buffer
+            .map_mut::<_, _, Result<_, <PixmanRenderer as Renderer>::Error>>(
+                &cursor_state.framebuffer_exporter,
+                0,
+                0,
+                cursor_buffer_size.w as u32,
+                cursor_buffer_size.h as u32,
+                |mbo| {
+                    let plane_pixman_format = pixman::FormatCode::try_from(DrmFourcc::Argb8888).unwrap();
+                    let cursor_dst = pixman::Image::from_slice_mut(
+                        plane_pixman_format,
+                        mbo.width() as usize,
+                        mbo.height() as usize,
+                        unsafe {
+                            slice::from_raw_parts_mut(
+                                mbo.buffer_mut().as_mut_ptr() as *mut u32,
+                                mbo.buffer().len() / 4,
+                            )
+                        },
+                        mbo.stride() as usize,
+                        false,
+                    )
+                    .map_err(|_| PixmanError::ImportFailed)?;
+
+                    self.pixman_renderer.bind(PixmanRenderBuffer::from(cursor_dst))?;
+                    let mut frame = self.pixman_renderer.render(self.cursor_size, output_transform)?;
+                    frame.clear(
+                        [0f32, 0f32, 0f32, 0f32],
+                        &[Rectangle::from_loc_and_size((0, 0), self.cursor_size)],
+                    )?;
+                    frame.render_texture_from_to(
+                        &cursor_texture,
+                        src,
+                        dst,
+                        element.damage_since(scale, None).as_slice(),
```

Comment:
Using `damage_since` is probably unnecessary here, you can just use `&[Rectangle::from_loc_and_size((0,0), dst.size)]`